bindings/sundials4py/examples/arkode/ark_heat1D.py

```diff
@@ -0,0 +1,212 @@
+#!/usr/bin/env python3
+# -----------------------------------------------------------------
+# Programmer(s): Cody J. Balos
+# -----------------------------------------------------------------
+# SUNDIALS Copyright Start
+# Copyright (c) 2025, Lawrence Livermore National Security,
+# University of Maryland Baltimore County, and the SUNDIALS contributors.
+# Copyright (c) 2013-2025, Lawrence Livermore National Security
+# and Southern Methodist University.
+# Copyright (c) 2002-2013, Lawrence Livermore National Security.
+# All rights reserved.
+#
+# See the top-level LICENSE and NOTICE files for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+# SUNDIALS Copyright End
+# -----------------------------------------------------------------
+# This example is a copy of examples/arkode/C_serial/ark_heat1D.c,
+# but ported to Python to use sundials4py.
+#
+# The following test simulates a simple 1D heat equation,
+#    u_t = k*u_xx + f
+# for t in [0, 10], x in [0, 1], with initial conditions
+#    u(0,x) =  0
+# Dirichlet boundary conditions, i.e.
+#    u_t(t,0) = u_t(t,1) = 0,
+# and a point-source heating term,
+#    f = 1 for x=0.5.
+#
+# The spatial derivatives are computed using second-order
+# centered differences, with the data distributed over N points
+# on a uniform spatial grid.
+#
+# This program solves the problem with either an ERK or DIRK
+# method.  For the DIRK method, we use a Newton iteration with
+# the SUNLinSol_PCG linear solver, and a user-supplied Jacobian-vector
+# product routine.
+#
+# 100 outputs are printed at equal intervals, and run statistics
+# are printed at the end.
+# -----------------------------------------------------------------
+
+import numpy as np
+from sundials4py.core import *
+from sundials4py.arkode import *
+
+
+class Heat1DProblem:
+    def __init__(self, N, k):
+        self.N = N
+        self.k = k
+        self.dx = 1.0 / (N - 1)
+
+    def set_init_cond(self, yvec):
+        y = N_VGetArrayPointer(yvec)
+        y[:] = 0.0
+        return 0
+
+    def f(self, t, yvec, ydotvec):
+        N, k, dx = self.N, self.k, self.dx
+        Y = N_VGetArrayPointer(yvec)
+        Ydot = N_VGetArrayPointer(ydotvec)
+        Ydot[:] = 0.0
+        c1 = k / dx / dx
+        c2 = -2.0 * k / dx / dx
+        # Vectorized Laplacian
+        Ydot[1:-1] = c1 * Y[:-2] + c2 * Y[1:-1] + c1 * Y[2:]
+        # Dirichlet BCs
+        Ydot[0] = 0.0
+        Ydot[-1] = 0.0
+        # Point source
+        isource = N // 2
+        Ydot[isource] += 0.01 / dx
```

Comment:
Why is this divided by `dx`? (Same question applies to the C version of the code)

Reply:
Its divided by dx so that its a cell-averaged source term of 0.01 independent of the mesh spacing.